handlers.go

```diff
@@ -28,9 +28,17 @@ func handleServices(add func(Change), services map[string]*charm.ServiceSpec) ma
 		service := services[name]
 		// Add the addCharm record if one hasn't been added yet.
 		if charms[service.Charm] == "" {
+			series := service.Series
+			if series == "" {
+				serviceCharm := charm.MustParseURL(service.Charm)
```

Comment:
Just `u := charm.MustParseURL(service.Charm)`?
And please add a comment stating that this is safe because the bundle data is assumed to be already verified, and therefore this must be a valid charm URL.